src-cpp/KafkaConsumerImpl.cpp

```diff
@@ -63,6 +63,9 @@ RdKafka::KafkaConsumer *RdKafka::KafkaConsumer::create (RdKafka::Conf *conf,
   if (!(rk = rd_kafka_new(RD_KAFKA_CONSUMER, rk_conf,
                           errbuf, sizeof(errbuf)))) {
     errstr = errbuf;
+    // rd_kafka_new() takes ownership only if succeeds
+    if (rk_conf)
```

Comment:
Not needed, rk_conf is always set in this constructor.
Revert to unambigiuate intent.
Thanks

Reply:
Then there can be null pointer dereferention on line 

:44 `if (!confimpl->rk_conf_) {`

Since this is probably called only once during live time of application, I'am goint test `confimpl` on NULL and rise exception if requiaremts are not met.

Reply:
Not rise exception but return error.